packages/components/src/components/form/bal-time-input/bal-time-input.tsx

```diff
@@ -0,0 +1,321 @@
+
+  /**
+   * Emitted when a keyboard key has pressed.
+   */
+  @Event() balKeyPress!: EventEmitter<KeyboardEvent>
+
+  /**
+   * Emitted when the input has clicked.
+   */
+  @Event() balClick!: EventEmitter<MouseEvent>
+
+  @Listen('click', { capture: true, target: 'document' })
+  listenOnClick(event: UIEvent) {
+    inputListenOnClick(this, event)
+  }
+
+  private resetHandlerTimer?: NodeJS.Timer
+
+  @Listen('reset', { capture: true, target: 'document' })
+  resetHandler(event: UIEvent) {
+    const formElement = event.target as HTMLElement
+    if (formElement?.contains(this.el)) {
+      inputHandleReset(this, this.initialValue, this.resetHandlerTimer)
+    }
+  }
+
+  connectedCallback() {
+    this.debounceChanged()
+    attachComponentToConfig(this)
+    this.initialValue = this.value || ''
+  }
+
+  componentDidLoad() {
+    this.inputValue = this.value
+  }
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAttributes(this.el, ['aria-label', 'tabindex', 'title'])
+  }
+
+  disconnectedCallback() {
+    detachComponentToConfig(this)
+  }
+
+  configChanged(state: BalConfigState): void {
+    this.language = state.language
+    this.region = state.region
+
+    if (!this.hasFocus && this.nativeInput) {
+      this.nativeInput.value = this.getFormattedValue()
+    }
+  }
+
+  /**
+   * Sets focus on the native `input`. Use this method instead of the global
+   * `input.focus()`.
+   */
+  @Method()
+  async setFocus() {
+    inputSetFocus(this)
+  }
+
+  /**
+   * Sets blur on the native `input`. Use this method instead of the global
+   * `input.blur()`.
+   * @internal
+   */
+  @Method()
+  async setBlur() {
+    inputSetBlur(this)
+  }
+
+  /**
+   * Returns the native `<input>` element used under the hood.
+   */
+  @Method()
+  async getInputElement() {
+    return this.nativeInput!
+  }
+
+  private getAllowedKeys() {
+    return [...NUMBER_KEYS, ...ACTION_KEYS, ':']
+  }
+
+  private getRawValue(): string {
+    const value = (this.value || '').toString()
+    return value
+  }
+
+  private getFormattedValue(): string {
+    return this.getRawValue()
+  }
+
+  private onInput = (ev: InputEvent) => {
+    const input = getInputTarget(ev)
+
+    if (input) {
+      this.inputValue = input.value
+    }
+
+    this.balInput.emit(this.inputValue)
+  }
+
+  private onFocus = (event: FocusEvent) => {
+    inputHandleFocus(this, event)
+  }
+
+  private onBlur = (ev: FocusEvent) => {
+    inputHandleBlur(this, ev)
+
+    const input = ev.target as HTMLInputElement | null
+    if (input) {
+      input.value = this.getFormattedValue()
+      inputHandleChange(this)
+    }
+  }
+
+  private onKeydown = (event: KeyboardEvent) => {
+    if (!isNil(event) && !isCtrlOrCommandKey(event)) {
+      if (!this.getAllowedKeys().includes(event.key)) {
+        return stopEventBubbling(event)
+      }
+    }
+  }
+
+  private onClick = (event: MouseEvent) => inputHandleClick(this, event)
```

Comment:
- [ ] please remove balClick and the onClick logic. we do not support that anymore